### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by Innocent259**
In this project milestone we: 

1. Removed current user method in the application controller
2. Installed and set up devise:

- [ ] -     Can register a new user.
- [ ] -     User logs in with a combination of email and password.
- [ ] -     Hashed password should be stored in the database.
- [ ] -     Ask for confirmation of email.
- [ ] -     Can reset password

3. Added CSS style to sign_up and sign_in devise views

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,13 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (Abenezer-Tilahun) [inline app/models/user.rb]: - [ ] As stated in the project requirement application should asks for confirmation of email. When I try to Login with combination of email and password the application doesn't ask me for the confirmation of email so Kindly Fix the issue.
- The application sign in without ask me for confirmation of email as shown with the below screenshhot.
image

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        <%= f.check_box :remember_me %>
+        <%= f.label :remember_me %>
+      </div>
+    <% end %>
+
+    <div class="actions">
+      <%= f.submit "Log in" %>
+    </div>
+  <% end %>
+<div>
```
Comment (Abenezer-Tilahun) [inline app/views/devise/sessions/new.html.erb]: - [ ] As stated in the project requirement you can [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.
- So kindly I recommend you to add Link to "Sign up", "Reset Password" and "Send confirmation of email." below the "Login " Link that enable the user to access the pages and increase users accessibility . Just Like the below 
image

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    <div class="field">
+      <%= f.label :password_confirmation %><br />
+      <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+    </div>
+
+    <div class="actions">
+      <%= f.submit "Sign up" %>
+    </div>
+  <% end %>
```
Comment (Abenezer-Tilahun) [inline app/views/devise/registrations/new.html.erb]: - [ ] As stated in the project requirement you can [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.
- So kindly I recommend you to add Link to "Login", and "Send confirmation of email." below the "Sign Up " Link that enable the user to access the pages and increase users accessibility . Just Like the below 
image

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @Innocent259 @maths29 

Great job for implemented  the some of the Project Requirement  , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages.
✔️ Correctly Install and setup devise .
✔️ Correctly allows for registering a new user, user can log in with a combination of email and password.
✔️ Good job for Implemented Blog app: add devise


almost

## Required Changes ♻️
_Check the comments below the review._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by topeogunleye — APPROVED** [review]
### Hi @Innocent259, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Added confirmation of password :heavy_check_mark: 
- Add Link to "Sign up", "Reset Password" and "Send confirmation of email." to  enable the user to access the pages and increase users accessibility :heavy_check_mark: 
 
## [Optional] suggestions:
👉 One optional suggestion for improvement is to follow the convention for naming your branch. Usually, branch names should use the past tense verb to indicate that the feature has been implemented. For example, instead of `add-devise`, you could use `added-devise`. This would make your branch name more consistent and clear for future pull requests. 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**